Source/artoolkitX.api.js

```diff
@@ -1,776 +1,778 @@
+      trackableId = artoolkitXjs.addTrackable(`${trackableObj.trackableType};${fileName}`)
+    }
 
-    /**
+    if (trackableId >= 0) {
+      this.trackables.push({ trackableId: trackableId, transformation: [], visible: false })
+      if (!this.userSetPatternDetection) { this[_updateDetectionMode]() }
+      return trackableId
+    }
+    throw new Error('Faild to add Trackable: ' + trackableId)
+  }
+
+  /**
      * Populates the provided float array with the current transformation for the specified marker. After
-     * a call to detectMarker, all marker information will be current. Marker transformations can then be
+     * a call to process, all marker information will be current. Marker transformations can then be
      * checked.
-     * @param {number} trackableUID	The unique identifier (UID) of the marker to query
-     * @return	{Float32Array} The dst array.
+     * @param {number} trackableUID The unique identifier (UID) of the marker to query
+     * @return {Float32Array} The dst array.
      */
+  getTransMatSquare (trackableUID) {
+    return this[_queryTrackableVisibility](trackableUID)
+  };
+
+  // /**
+  //  * T.B.: Make this function static as it is a util function and doesn't manipulate the ARController object (this)
+  //  *
+  //  Converts the given 3x4 marker transformation matrix in the 12-element transMat array
+  //  into a 4x4 WebGL matrix and writes the result into the 16-element glMat array.
+
+  //  If scale parameter is given, scales the transform of the glMat by the scale parameter.
+
+  //  @param {Float64Array} transMat The 3x4 marker transformation matrix.
+  //  @param {Float64Array} glMat The 4x4 GL transformation matrix.
+  //  @param {number} [scale] The scale for the transform.
+  // */
+  // ARController.transMatToGLMat = function(transMat, glMat, scale) {
+  //     if(glMat == undefined){
+  //         glMat = new Float64Array(16);
+  //     }
+  //  glMat[0 + 0*4] = transMat[0]; // R1C1
+  //  glMat[0 + 1*4] = transMat[1]; // R1C2
+  //  glMat[0 + 2*4] = transMat[2];
+  //  glMat[0 + 3*4] = transMat[3];
+  //  glMat[1 + 0*4] = transMat[4]; // R2
+  //  glMat[1 + 1*4] = transMat[5];
+  //  glMat[1 + 2*4] = transMat[6];
+  //  glMat[1 + 3*4] = transMat[7];
+  //  glMat[2 + 0*4] = transMat[8]; // R3
+  //  glMat[2 + 1*4] = transMat[9];
+  //  glMat[2 + 2*4] = transMat[10];
+  //  glMat[2 + 3*4] = transMat[11];
+  //  glMat[3 + 0*4] = 0.0;
+  //  glMat[3 + 1*4] = 0.0;
+  //  glMat[3 + 2*4] = 0.0;
+  //  glMat[3 + 3*4] = 1.0;
+  //  if (scale != undefined && scale !== 0.0) {
+  //   glMat[12] *= scale;
+  //   glMat[13] *= scale;
+  //   glMat[14] *= scale;
+  //  }
+  //  return glMat;
+  // };
+
+  /**
+  * Returns the projection matrix computed from camera parameters for the ARController.
+  *
+  * @return {Float32Array} The 16-element WebGL camera matrix for the ARController camera parameters.
+  */
+  getCameraMatrix (nearPlane = 0.1, farPlane = 1000) {
```

Comment:
@ThorstenBux we have getProjectionMatrix() 
https://github.com/augmentmy-world/artoolkitx/blob/4b707183779b7ff0cf9434620eb4f2f284f11219/Source/artoolkitx.js/ARX_additions.js#L12  ARX_additions.js that is exactly the same.

Reply:
that is, we can call 
`return artoolkitXjs.getProjectionMatrix(nearPlane, farPlane)`

Reply:
Right, good catch. 
From what I recall of Emscripten the direct call to the C-function should be more performant than the ccall function. Also, I would like to keep the name as getCameraMatrix because it is easier understandable (for me) and 3rd thing is that the ARX_additions.js is another thing to think about during compilation so if we can we should probably try to eliminate it in the future.

Reply:
> From what I recall of Emscripten the direct call to the C-function should be more performant than the ccall function. 
i will re-read the emscripten docs, don't know about this.
>  I would like to keep the name as getCameraMatrix because it is easier understandable (for me) 
Yes, for me too.
I see also that in `ARX_additions.js` there are other functions not used  for example `loadOpticalParams`. Maybe we should solve this problem at another time, in another issue in the future.

Reply:
Yes, I'm not saying to remove ARX_additions.js now. That would be something to evaluate in the future. (I can't recall anymore why I added it, but I know it was a learning process and at some point having ARX_additions seems the way to go). For the case at hand, I vote for having the function inside the API script.

Reply:
>  I'm not saying to remove ARX_additions.js now. 

yes completely agree!

> For the case at hand, I vote for having the function inside the API script.

so to be clear: 
- first option: we leave as it is.
- second option:  we trasform it into:

```js
getCameraMatrix (nearPlane = 0.1, farPlane = 1000) {
return artoolkitXjs.getProjectionMatrix(nearPlane, farPlane)
}
```

Reply:
Leave in the API as is and remove the function from ARX_additions.

Reply:
ok i get it!

Reply:
i will prepare a PR in the emscripten branch of https://github.com/augmentmy-world/artoolkitx